app/controllers/comments/votes_controller.rb

```diff
@@ -0,0 +1,38 @@
+module Comments
+  class VotesController < ApplicationController
+    load_and_authorize_resource :comment
+    before_action :authenticate_user!, only: [:create, :destroy]
+    before_action :verify_comments_open!, only: [:create, :destroy]
```

Comment:
I moved the `before_actions` under the `load_and_authorize_resource` because in the `verify_comments_open!` method we need to have the `@comment`.

So I had to decide between split the `before_actions` and move the `load_and_authorize_resource` in the middle or keep it as you have seen.

For now I'm going to follow the convention on the order between the `authenticate_user!` and the `load_and_authorize_resource` and move the `veriry_comments_open!` to the end.

Reply:
The thing is, it's very confusing to first authorize the comment and then authenticate the user. I'm going to check it.